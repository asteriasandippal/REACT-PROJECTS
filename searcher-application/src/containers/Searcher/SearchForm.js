import React from 'react';
import Button from '../../components/Button';
import Input from '../../components/Input';

class SearchForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            query: ''
        }

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(e) {
        e.preventDefault();
        this.props.onSearch(this.state.query);
    }

    render() {
        return (
            <div>
                <form className="pt-4 pb-4" onSubmit={this.handleSubmit}>
                    <div className="input-group mb-3">
                        <Input 
                            className="form-control-lg"
                            placeholder="Search phots..."
                            onChange={(e) => this.setState({ query: e.target.value })}
                        />
                        <div className="input-group-append">
                            <Button 
                                text="Search" 
                                className="btn-lg"
                                onClick={this.handleSubmit}
                            />
                        </div>
                    </div>
                </form>
            </div>
        );
    }
}

export default SearchForm;
