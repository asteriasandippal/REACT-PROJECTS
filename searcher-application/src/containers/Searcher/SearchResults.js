import React from 'react';
import GridBlock from '../../components/GridBlock';
import Card from '../../components/Card';

class SearchResults extends React.Component {
    render() {
        const { showPhotos } = this.props;
        return (
            <div className="row">
                {showPhotos.map(photo =>
                    <GridBlock colume="4" key={photo.id}> 
                        <Card 
                            image={photo.image_url[0]} 
                            name={photo.name}
                            description={photo.description}
                        />
                    </GridBlock>
                )}
            </div>
        );
    }
}

export default SearchResults;
