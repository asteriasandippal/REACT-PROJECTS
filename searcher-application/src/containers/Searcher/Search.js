import React from 'react';
import SearchForm from './SearchForm';
import SearchResults from './SearchResults';

class Search extends React.Component {
    constructor(props) {
        super(props);
        this.state= {
            photos: []
        }
        this.onSearching = this.onSearching.bind(this);
    }

    onSearching(query) {
        console.log(query);
        const apiUrl = `https://api.500px.com/v1/photos/search?consumer_key=u4c2ZyL3DxwqagztgOjYT5K1ii1u72ILA0zfoyBE&image_size[]=3&image_size[]=4&term=${query}`
        fetch(apiUrl)
            .then(response => response.json())
            .then(data => this.setState({photos : data.photos}))
            .catch(error => {console.log(error)})
    }

    render() {
        return (
            <div>
                <SearchForm onSearch={this.onSearching} />
                <SearchResults showPhotos={this.state.photos}/>
            </div>
        );
    }
}

export default Search;
