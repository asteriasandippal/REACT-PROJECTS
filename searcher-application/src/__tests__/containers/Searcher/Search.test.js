require('isomorphic-fetch');

import React from 'react';
import { shallow } from 'enzyme';

import Search from '../../../containers/Searcher/Search';
import SearchForm from '../../../containers/Searcher/SearchForm';
import SearchResults from '../../../containers/Searcher/SearchResults';

test('Search render correctly', () => {
    const wrapper = shallow(
        <Search/>
    );

    const tree = wrapper.getElement();
    expect(tree).toMatchSnapshot();
});

test('Search Searches correctly', () => {
    const wrapper = shallow(
        <Search/>
    );

    const searchForm = wrapper.find(SearchForm);
    searchForm.prop('onSearch')('test');

    const tree = wrapper.getElement();
    expect(tree).toMatchSnapshot(); 
});

