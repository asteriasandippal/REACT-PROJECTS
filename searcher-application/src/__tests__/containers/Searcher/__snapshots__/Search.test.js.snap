// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Searches correctly 1`] = `
<div>
  <SearchForm
    onSearch={[Function]}
  />
  <SearchResults
    showPhotos={Array []}
  />
</div>
`;

exports[`Search render correctly 1`] = `
<div>
  <SearchForm
    onSearch={[Function]}
  />
  <SearchResults
    showPhotos={Array []}
  />
</div>
`;
